src/bootstrap/tool.rs

```diff
@@ -33,6 +33,44 @@ struct ToolBuild {
     allow_features: &'static str,
 }
 
+fn tooling_output(
+    mode: Mode,
+    tool: &str,
+    stage: u32,
```

Comment:
nit: because you're doing math on it, please name it `build_stage`. Then in the Rustdoc Step, you can pass in build_compiler.stage instead of doing math on target_compiler.stage.